### PR TITLE
Initial stat() docs

### PR DESCRIPTION
**Pull request by therealsilence**
Document Storm stat() operator

### CODE REVIEW

File: docs/synapse/userguides/ug017_storm_ref_stats.rst [hunk 1]
```diff
@@ -1,16 +1,192 @@
-Storm Reference - Statistical Operators
-=======================================
+Storm Reference - Statistical Operator
+======================================
 
-Todo
+The statistical operator is used to generate data about data in Synapse.
+
+``Stat()`` is defined in common.py_ as opposed to storm.py_.
```
Comment (vEpiphyte) [inline docs/synapse/userguides/ug017_storm_ref_stats.rst]: `Stat` vs `stat` consistency. we should use the lowercase form.

File: docs/synapse/userguides/ug017_storm_ref_stats.rst [hunk 2]
```diff
@@ -1,16 +1,192 @@
...
+  **stat(** *<handler>* **,** *<prop>* [ **,valu =** *<valu>* ] **)**
+
+**Macro Syntax:**
+
+N/A
+
+**Examples:**
+
+**count** - returns the number of nodes with the specified *<prop>* or *<prop> / <valu>*
+
```
Comment (vEpiphyte) [inline docs/synapse/userguides/ug017_storm_ref_stats.rst]: Can we add an example showing the count of `tufo:form` values to represent the number of nodes in a graph?

File: docs/synapse/userguides/ug017_storm_ref_stats.rst [hunk 3]
```diff
@@ -1,16 +1,194 @@
-Storm Reference - Statistical Operators
-=======================================
+Storm Reference - Statistical Operator
+======================================
 
-Todo
+The statistical operator is used to generate data about data in Synapse.
```
Comment (vertexmc) [inline docs/synapse/userguides/ug017_storm_ref_stats.rst]: I would change `generate data about data` to something about how it returns calculated statistics about the data.

File: docs/synapse/userguides/ug017_storm_ref_stats.rst [hunk 4]
```diff
@@ -1,16 +1,194 @@
...
+
+``stat()`` is defined in common.py_ as opposed to storm.py_.
+
+``stat()`` operates differently from other Storm operators:
+
+* ``stat()`` operates directly on the Synapse storage layer using the row-level APIs (as opposed to the node (form) APIs used by other Storm operators). This is an optimization that allows ``stat()`` to answer questions across large data sets ("all of the IP addresses in Synapse") that would otherwise be too "heavy" (non-performant) to lift.
+
+* Depending on the specific ``stat()`` handler used and the optimizations available in a particular Syanpse storage backing, the amount of time for a given ``stat()`` query to return may vary. For example, "count" operations will generally return much faster than "min" or "max" operations, even with the use of row-level APIs.
```
Comment (vertexmc) [inline docs/synapse/userguides/ug017_storm_ref_stats.rst]: Synapse typo

File: docs/synapse/userguides/ug017_storm_ref_stats.rst [hunk 5]
```diff
@@ -1,16 +1,194 @@
...
+
+* ``stat()`` is designed as a stand-alone operator; because it uses a different set of APIs, it cannot operate on the output of a previous Storm query and so cannot be "chained" as part of a larger query.
```
Comment (vertexmc) [inline docs/synapse/userguides/ug017_storm_ref_stats.rst]: This is really important to call out. What happens if you chain it together anyway?

File: docs/synapse/userguides/ug017_storm_ref_stats.rst (hunk 3 above)
Comment (therealsilence) [reply docs/synapse/userguides/ug017_storm_ref_stats.rst]: fixed

File: docs/synapse/userguides/ug017_storm_ref_stats.rst (hunk 4 above)
Comment (therealsilence) [reply docs/synapse/userguides/ug017_storm_ref_stats.rst]: fixed

File: docs/synapse/userguides/ug017_storm_ref_stats.rst (hunk 5 above)
Comment (therealsilence) [reply docs/synapse/userguides/ug017_storm_ref_stats.rst]: Based on quick and dirty testing it looks like Synapse simply ignores the stat() operator wherever it appears in the query (middle, end) but executes the other query elements as if the stat() operator were not present.

**Pull request merged**